### PR TITLE
mgmt, local generation ComputeRP 2023-09-01 and GalleryRP 2022-08-03

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [PR#4821](https://github.com/Azure/sdk-release-request/issues/4821)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java [hunk 1]
```diff
@@ -2130,7 +2130,7 @@ public Accepted<VirtualMachine> beginCreate() {
                         .manager()
                         .serviceClient()
                         .getVirtualMachines()
-                        .createOrUpdateWithResponseAsync(resourceGroupName(), vmName, innerModel())
+                        .createOrUpdateWithResponseAsync(resourceGroupName(), vmName, innerModel(), null, null)
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java:2133]: what is the 2 parameters?

File: sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,14 @@
 ### Features Added
 
 - Supported `userData` for `VirtualMachine` in create and update.
+- Fixed `actual and formal argument lists differ in length` exception in `VirtualMachineInner` class.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md:8]: remove this line, this does not relate to customer.

File: sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,14 @@
...
+
+### Other Changes
+
+#### Dependency Updates
+
+- Updated `api-version` of `ComputeRP` to `2023-09-01`.
+- Updated `api-version` of `GalleryRP` to `2022-08-03`.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md:15]:
```suggestion
- Updated `api-version` to `2023-09-01`.
```

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java (hunk 1 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java:2133]: The two parameters `ifMatch` and `ifNoneMatch` were added in method `VirtualMachinesClient.createOrUpdateWithResponseAsync`.

File: sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md (hunk 3 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md (hunk 2 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java:2133]: OK. So they started to support etag...

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**